### PR TITLE
Allow tenants with suffix _ and digits MODEUR-130

### PR DESCRIPTION
**Pull request by adamdickmeiss**
This PR is for 1.0 series only (where folio-vertx-lib was not split out).

### CODE REVIEW

File: src/main/java/org/folio/tlib/util/TenantUtil.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
   // PostgreSQL names are case insensitive and must not start with a digit.
   // The maximum length is 63 characters, schema = tenant + '_' + moduleName
   // where tenant and moduleName length have a maximum of 31 each.
-  private static final String TENANT_PATTERN_STRING = "^[a-z][a-z0-9]{0,30}$";
+  private static final String TENANT_PATTERN_STRING = "^[a-z][a-z0-9]*(_[0-9]+)*$";
```
Comment (julianladisch) [inline src/main/java/org/folio/tlib/util/TenantUtil.java:10]: It should accept every tenant that Okapi and PostgreSQL currently accept:
`"^[_a-z][_a-z0-9]*$"`

More restriction can be imposed at a later time.

File: src/main/java/org/folio/tlib/util/TenantUtil.java [hunk 2]
```diff
@@ -27,6 +27,10 @@ public static String tenant(RoutingContext ctx) {
       throw new IllegalArgumentException(
           "X-Okapi-Tenant header must match " + TENANT_PATTERN_STRING);
     }
+    if (tenant.length() > 31) {
+      throw new IllegalArgumentException(
+          "X-Okapi-Tenant header size must not be greater than 31");
+    }
```
Comment (julianladisch) [inline src/main/java/org/folio/tlib/util/TenantUtil.java:33]: Okapi and PostgreSQL currently accept longer tenant ids.

The length restriction can be added later.

File: src/main/java/org/folio/tlib/util/TenantUtil.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/tlib/util/TenantUtil.java:10]: OK

File: src/main/java/org/folio/tlib/util/TenantUtil.java (hunk 2 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/tlib/util/TenantUtil.java:33]: OK

**Review by julianladisch — CHANGES_REQUESTED** [review]
Okapi and PostgreSQL currently accept tenant ids that are longer than 31 characters.

Therefore limiting the length to 31 characters should be removed from this pull request.

The length restriction can be added later.

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Pull request merged**